libs/designer-ui/src/lib/html/plugins/toolbar/Toolbar.tsx

```diff
@@ -199,13 +225,62 @@ export const Toolbar = ({ readonly = false }: toolbarProps): JSX.Element => {
         <img className={'format'} src={isInverted ? clockWiseArrowDark : clockWiseArrowLight} alt={'clockwise arrow'} />
       </button>
       <Divider />
-      <BlockFormatDropDown disabled={readonly} blockType={blockType} editor={editor} />
-      <FontDropDown fontDropdownType={FontDropDownType.FONTFAMILY} value={fontFamily} editor={editor} disabled={readonly} />
-      <FontDropDown fontDropdownType={FontDropDownType.FONTSIZE} value={fontSize} editor={editor} disabled={readonly} />
+      <BlockFormatDropDown disabled={formattingButtonsDisabled} blockType={blockType} editor={editor} />
+      <FontDropDown
+        fontDropdownType={FontDropDownType.FONTFAMILY}
+        value={fontFamily}
+        editor={editor}
+        disabled={formattingButtonsDisabled}
+      />
+      <FontDropDown fontDropdownType={FontDropDownType.FONTSIZE} value={fontSize} editor={editor} disabled={formattingButtonsDisabled} />
       <Divider />
-      <Format activeEditor={activeEditor} readonly={readonly} />
+      <Format activeEditor={activeEditor} readonly={formattingButtonsDisabled} />
       <ListPlugin />
       <LinkPlugin />
+      {setIsRawText ? (
+        <button
+          aria-label="Raw code toggle"
+          className={css('toolbar-item', isRawText && 'active')}
+          disabled={readonly}
```

Comment:
There is not yet support for disabling this button if Lexical would fail to parse the raw HTML provided by the user.